syntax = "proto3";

package envoy.type;

option java_package = "io.envoyproxy.envoy.type";
option java_outer_classname = "RangeProto";
option java_multiple_files = true;

// [#protodoc-title: Range]

// Specifies the int64 start and end of the range using half-open interval semantics [start,
// end).
message Int64Range {
  // start of the range (inclusive)
  int64 start = 1;

  // end of the range (exclusive)
  int64 end = 2;
}

// Specifies the double start and end of the range using half-open interval semantics [start,
// end).
message DoubleRange {
  // start of the range (inclusive)
  double start = 1;

  // end of the range (exclusive)
  double end = 2;
}
